circus-train-aws-sns/README.md

```diff
@@ -53,28 +70,110 @@ Currently the S3 credentials specified in the JCEKS key store are used to authen
+	  "replicaTable" : "replicaDb.replicaTable",
+	  "replicaTableLocation" : "s3://bucket/path",
+	  "replicaMetastoreUris" : "thrift://host:9083",
+	  "bytesReplicated" : 84837488
+	}
+
+#### Replication success for a partitioned table
+The following shows an example JSON message representing a successful replication of a non-partitioned table 
+and also the usage of custom "pipeline-id" header that was set using the `sns-event-listener.headers` 
+configuration value described above:
+
+	{
+	  "protocolVersion" : "2.0.0",
+	  "type" : "SUCCESS",
+	  "headers" : {
+	    "pipeline-id" : "0943879438"
+	  },
+	  "startTime" : "2016-06-01T15:27:38.365Z",
+	  "endTime" : "2016-06-01T15:27:39.000Z",
+	  "eventId" : "ctp-20160601T152738.363Z-CzbZaYfj",
+	  "sourceCatalog" : "sourceCatalogName",
+	  "replicaCatalog" : "replicaCatalogName",
+	  "sourceTable" : "srcDb.srcTable",
+	  "replicaTable" : "replicaDb.replicaTable",
+	  "replicaTableLocation" : "s3://bucket/path",
+	  "replicaMetastoreUris" : "thrift://host:9083",
+	  "partitionKeys" : {
+	    "local_date" : "string",
+	    "local_hour" : "int"
+	  },
+	  "modifiedPartitions" : [ [ "2014-01-01", "0" ], [ "2014-01-01", "1" ] ],
+	  "bytesReplicated" : 84837488
+	}
+
+#### Replication failure
+The following shows an example JSON message representing the failure of a table replication:
 
 	{
-	   "protocolVersion" : "1.0",
-	   "type" : "FAILURE",
-	   "headers" : {
-	      "pipeline-id" : "0943879438"
-	   },
-	   "eventId" : "ctp-20160601T152738.363Z-CzbZaYfj",
-	   "startTime" : "2016-06-01T15:27:38.365Z",
-	   "sourceCatalog" : "sourceCatalogName",
-	   "sourceTable" : "srcDb.srcTable",
-	   "replicaCatalog" : "replicaCatalogName",
-	   "replicaTable" : "replicaDb.replicaTable",
-	   "bytesReplicated" : 84837488,
-	   "modifiedPartitions" : [
-	      ["2014-01-01", "0"],
-	      ["2014-01-01", "1"]
-	   ],
-	   "endTime" : "2016-06-01T15:27:38.365Z",
-	   "errorMessage" : "error message"
+	  "protocolVersion" : "2.0.0",
+	  "type" : "FAILURE",
+	  "startTime" : "2016-06-01T15:27:38.365Z",
+	  "endTime" : "2016-06-01T15:27:39.000Z",
+	  "eventId" : "ctp-20160601T152738.363Z-CzbZaYfj",
+	  "sourceCatalog" : "sourceCatalogName",
+	  "replicaCatalog" : "replicaCatalogName",
+	  "sourceTable" : "srcDb.srcTable",
+	  "replicaTable" : "replicaDb.replicaTable",
+	  "replicaTableLocation" : "s3://bucket/path",
+	  "replicaMetastoreUris" : "thrift://host:9083",
+	  "errorMessage" : "Connection timed out"
 	}
 
 ## Limitations
-SNS messages have a size limit of 256KB. It is possible that you could exceed this bound if your replication touches a very large number of partitions. In this case your message will not be sent and instead logged with a warning.
\ No newline at end of file
+SNS messages have a size limit of 256KB. It is possible that you could exceed this bound if your replication touches a very large number of partitions. In this case your message will not be sent and instead logged with a warning.
```

Comment:
So you could have a SUCCESS with a large message that cannot be sent and will be logged, would that then fail the replication?
Trying to figure how a user is supposed to find out if something like this happens. But looks like reading the logs is the only way.

Reply:
I don't think we'll fail the replication so yes, they'd have to look in the logs.